### PR TITLE
Homepage

### PR DESCRIPTION
**Pull request by Tanay-J**
This PR deals with
- React router setup, Public and Private routes 
- Landing page and required components, api-requests and css modules
- Custom hook for local storage and dark theme support using it

### Preview
https://booklore-video-library-3732y9bgj-tanay-j.vercel.app/

### CODE REVIEW

**Review by AnuragSahu11 — APPROVED** [review]
Very nice UI , good to see different component for public routes and private routes, you can merge

**Review by vermasavita — APPROVED** [review]
Looks good to merge.

File: src/App.js [hunk 1]
```diff
@@ -1,7 +1,13 @@
 import "./App.css";
+import { PrivateRoutes, PublicRoutes } from "./utils/routes";
 
 function App() {
-  return <div></div>;
+  return (
+    <div>
+      <PublicRoutes />
```
Comment (bhavyaj12) [inline src/App.js:7]: Great use of public and private routes

File: src/components/homepage/Featured.jsx [hunk 2]
```diff
@@ -1,4 +1,15 @@
+import { getVideos } from "../../utils/service-requests/video-services";
+import { VideoCard } from "../listing";
+import styles from "./homepage.module.css";
+
 const Featured = () => {
-  return <></>;
+  const { videoList } = getVideos();
+  return (
+    <section className={`${styles.videolist_container}`}>
```
Comment (bhavyaj12) [inline src/components/homepage/Featured.jsx:8]: Nice use of slice function

File: src/components/listing/VideoCard.jsx [hunk 3]
```diff
@@ -0,0 +1,48 @@
+import styles from "./listing.module.css";
+import { BsFillPlayCircleFill, BsThreeDotsVertical } from "react-icons/bs";
+import { Link } from "react-router-dom";
```
Comment (bhavyaj12) [inline src/components/listing/VideoCard.jsx]: Try to follow the convention of global imports first and then the local imports

File: src/components/listing/listing.module.css [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+}
+ul {
+  list-style-type: none;
+  padding: 4px;
+}
+.options {
+  position: absolute;
+  right: 5%;
+  bottom: 0px;
+  background-color: rgba(66, 65, 65, 0.9);
```
Comment (bhavyaj12) [inline src/components/listing/listing.module.css:53]: Try to put colors in your global css as a variable, would make it cleaner

File: src/components/navigation/Navbar.jsx [hunk 5]
```diff
@@ -0,0 +1,54 @@
+import { BsMoonFill, BsPlayCircleFill, BsSun } from "react-icons/bs";
+import { useEffect } from "react";
+import styles from "./navigation.module.css";
+import { Link } from "react-router-dom";
+import { useLocalStorage } from "../../utils/hooks/useLocalStorage";
+
```
Comment (bhavyaj12) [inline src/components/navigation/Navbar.jsx:6]: you can add a jsconfig.json file to your global folder, where your env file is basically. There in the baseurl if you include src folder in compiler options then you can avoid 
.. /..  Everywhere

File: src/pages/homepage/Homepage.jsx [hunk 6]
```diff
@@ -0,0 +1,17 @@
+import { Banner, Featured } from "../../components/homepage";
```
Comment (bhavyaj12) [inline src/pages/homepage/Homepage.jsx]: You can create an import export index.js file in the components folder and if you use jsconfig.json then the path can be shortened to just components
Suggestion for future projects, if possible

File: src/utils/hooks/useLocalStorage.js [hunk 7]
```diff
@@ -0,0 +1,14 @@
+import { useEffect, useState } from "react";
+
+const useLocalStorage = (key, defaultValue) => {
```
Comment (bhavyaj12) [inline src/utils/hooks/useLocalStorage.js:3]: Really nice to create this for localstorage

**Review by bhavyaj12 — APPROVED** [review]
Great work. Code is really well written.
Have left some minor suggestions.
Also it's better that you have featured category cards in home page as well, as it's mentioned. And there's an api to fetch categories.
UI is clean and good.

File: src/components/listing/listing.module.css (hunk 4 above)
Comment (Tanay-J) [reply src/components/listing/listing.module.css:53]: It's one off usage, so didn't create a variable. But I get your point, a variable would look cleaner.

File: src/components/navigation/Navbar.jsx (hunk 5 above)
Comment (Tanay-J) [reply src/components/navigation/Navbar.jsx:6]: This is a new learning for me, I will surely incorporate in the project. Thanks for sharing.

**Pull request merged**